### DISCUSSION TITLE
Accessing FIBO In A Corporate Firewalled Environment

### DISCUSSION THREAD

**Discussion by carmenchui** [edmcouncil/fibo#1362] [General] [answered]
Hi FIBO Development Team,

I am not sure if this problem is unique to FIBO or Protege, but I am working with a Canadian banking corporation that wishes to examine FIBO through various tools, such as Protege Desktop, Stardog, and other tools listed online (http://owl.cs.manchester.ac.uk/tools/list-of-reasoners/ and https://www.uni-ulm.de/en/in/ki/research/software/).

However, the corporation has a firewall in place and prevents Protege from loading imported terminology from OMG and W3: Protege constantly prompts me to 'resolve' the imports when trying to load FIDO PROD and DEV.  This occurs regardless of whether I am using the quick start files, or the downloaded zip file of the ontology.

An example import from FIBO that fails inside the firewall even if I try loading via URL: https://www.omg.org/spec/LCC/Languages/LanguageRepresentation/ (This is imported by https://spec.edmcouncil.org/fibo/ontology/master/2020Q2/FND/Utilities/AnnotationVocabulary/)

I have tried setting the Java proxy settings in the Protege launcher but the result is the same.  I have logged this issue with the Protege team as well (https://github.com/protegeproject/protege/issues/967).

Have you come across any other FIBO users with this issue in corporate environments?  For context, I am using a corporate laptop when using Protege Desktop and a Linux virtual machine on the corporation's internal network when trying to use command-line OWL/DL reasoning tools to analyze FIBO.

Any help or guidance would be appreciated.

Thank you,

Carmen

**Comment by mereolog** [comment]
Can you access these ontologies directly, for example, can you reach https://www.omg.org/spec/LCC/Countries/CountryRepresentation/ from your browser?

**Comment by carmenchui** [comment]
Hi @mereolog, the ontologies can be accessed directly from the browser, just not through Protege.

**Comment by mereolog** [comment]
Ok, you may then download all of them into a local folder and use them when Protege asks for import resolutions - if you use dev.fibo-quickstart.ttl there should be just two of them:

https://www.omg.org/spec/LCC/Countries/CountryRepresentation/
https://www.omg.org/spec/LCC/Languages/LanguageRepresentation/

I know that this is sort of laborious and fishy but I cannot think about any other solution now.
Perhaps other members of this forum will have better ideas.

**Comment by ElisaKendall** [comment, marked as answer]
Hi Carmen,

A number of banks, including Wells Fargo and others, have this issue, although we've been using a 'follow your nose' mechanism to date.  We should probably add a zip file that includes all of the important imports to the download page, although they do change from time to time (not often).  

The LCC ontologies are an Object Management Group (OMG) standard, currently on version 1.1, and likely updated later this year to cover New Macedonia :).  Those ontologies are available in the attached zip file.  
[LCC-20200710.zip](https://github.com/edmcouncil/fibo/files/4904083/LCC-20200710.zip).  Note that the load file is under the 1.1 folder in the zip file.  I zipped the current git repository, which is the 1.1 published version.

There are a few other files that you'll need - the OMG's specification metadata recommendation, which is in the attached zip - 
[SpecificationMetadata.zip](https://github.com/edmcouncil/fibo/files/4904239/SpecificationMetadata.zip).  You'll also need SKOS core, available from the W3C, also attached - 
[core.zip](https://github.com/edmcouncil/fibo/files/4904244/core.zip).  Finally, we reference, but don't import, the Dublin Core Metadata terms ontology, which is also attached - 
[terms.zip](https://github.com/edmcouncil/fibo/files/4904268/terms.zip)

I -think- this everything.  Hope this is what you were looking for.  Let us know if you need more assistance. 

Best,

Elisa

**Comment by carmenchui** [comment]
Hi Elisa and Pawel,

Thank you for your quick responses!
I'll give this a try and let you know if any additional assistance is required.

Much appreciated,

Carmen

**Comment by trypuz** [comment]
Carmen (@carmenchui), let me know if you need any further assistance.

**Comment by carmenchui** [comment]
Hi Robert @trypuz,

I'm still waiting to hear back from a colleague but did have a question.

Would it actually be easier to load FIBO by first using the "Gather Ontologies" feature in Protege on a personal computer, zip up the files, and then reload these files on the corporate computer?

The reason why I ask is that the Protege cache retains which ontologies have been prompted for a local file -- if the user has opted not to resolve the imports, the program will not prompt the user again.  I've recommended other users to clear their Protege cache every time they try to load FIBO on a corporate computer, but it does get tedious after a while.

**Comment by jgeluk** [comment]
@carmenchui did you add your company's proxy details into the protege start script?

- https://mailman.stanford.edu/pipermail/protege-user/2016-September/005276.html

**Comment by carmenchui** [comment]
Hi @jgeluk, yes, I added the proxy details in the Protege start script but Protege still returns errors when importing the OMG and W3C imports.

**Comment by trypuz** [comment]
@carmenchui, if you still have a problem with FIBO imports, please provide us with more details concerning tools you are using (versions, platform, os, etc.).

**Comment by carmenchui** [comment]
Hi @trypuz, I just checked with another colleague and they were able to load FIBO manually on the corporate laptop in its entirety with files @ElisaKendall provided.

I tried loading FIBO just now on my corporate laptop with the follow-your-nose starting point URL (https://spec.edmcouncil.org/fibo/ontology/master/2020Q2/LoadFIBOProd.rdf) as well and it appears to load without issues.  However, I'm not sure if any of the imports are still being blocked by the corporate proxy/firewall.

What would be helpful would be for the FIBO team to also provide the Protege metrics (number of axioms, logical axiom count, classes, etc.) for each release because the colleague and I had to compare Protege metrics from a personal computer to make sure we got the same number of axioms/classes on the corporate machine.

Can I check with you that these are the correct metrics for FIBO 2020 Q2?

<img width="757" alt="Capture2" src="https://user-images.githubusercontent.com/4117565/91599663-a0965780-e934-11ea-8966-7db70dae63ee.PNG">


For reference, the corporate machine runs the following:
- Lenovo ThinkPad T470 (16GB RAM, Intel Core i5-6300U CPU)
- Windows 10 Enterprise version 1703 (OS Build 15063.2409);  64-bit, x64 processor
- Java SE Runtime Environment (build 1.8.0_24-b26)
- Protege 5.5.0 with the following fields filled out in the startup script (.bat): 
-Dhttp.proxyHost -Dhttp.proxyPort -Dhttp.proxyUser -Dhttp.proxyPassword
-Dhttps.proxyHost -Dhttps.proxyPort -Dhttps.proxyUser -Dhttps.proxyPassword

Unfortunately, I don't have any information on the corporate proxy, but if that's needed, I can try to find that out (I am not sure if I am privy to this information since I am not an employee but a researcher working with the corporation).

**Comment by trypuz** [comment]
Hi @carmenchui, it is actually a great idea to provide the Protege metrics for each release. I'll ask the FIBO Team to create a Jenkins job for it. Though we have to be careful here to not confuse our Community because we release a few FIBO products that may differ in their metrics, e.g., FIBO with metadata will have more triples than FIBO without metadata info. The same applies to the quickstart files.

**Comment by carmenchui** [comment]
Hi @trypuz, that makes sense.  Perhaps outlining which Protege metrics corresponds to which files (LoadFIBOProd, the quickstart) would also be helpful since the FIBO Team does provide various access points for its users.  I think providing the metrics would be especially useful for the corporate users just in case their firewall or proxy is blocking imports.

**Comment by karabshi** [comment]
hello @ElisaKendall  i'm also having the same issue with downloading FIBO behind a firewall and setting proxy didn't work. I tried to download the zip files  but i'm still getting the import issues. for some reason it's not reading the catalog files within the corresponding ontology directories. any clue what the issue may be? thank you.

**Comment by ElisaKendall** [comment]
@karabshi Hi, Did you grab the zip files attached above in my comment to Carmen?  And if so, does that fix the issue?

Otherwise I (or possibly Robert, depending on where you are located) should probably have a 1-1 debugging session with you to see what you are able to download and what is actually missing.  You probably should start with the Q4 2020 release so that we are on the same page.  If you have not downloaded that version, please do that first.

Best,

Elisa

**Comment by karabshi** [comment]
@ElisaKendall i did download the files you referenced and it's still not able to import. sure would be happy to go on a 1-1 with you to debug the issue.

**Comment by ElisaKendall** [comment]
@karabshi My email address is ekendall@thematix.com - please feel free to drop me a note and we can find a time.
